Stellar-ledger.x

```diff
@@ -412,14 +412,19 @@ struct ContractEvent
     body;
 };
 
+struct OperationEvents
+{
+    ContractEvent events<>;
+};
+
 struct TransactionMetaV3
 {
     LedgerEntryChanges txChangesBefore; // tx level changes before operations
                                         // are applied if any
     OperationMeta operations<>;         // meta for each operation
```

Comment:
@sisuresh What's the rationale for keeping the events out of operation meta? I thought I heard you say there's a benefit to keeping it at the tx level, but why is unclear to me.

Reply:
The rationale was that we'd like to hash events separately from OperationMeta (which includes the ledger entry changes for the operation). If a user wants to verify the authenticity of an event, they would hash all `events` in the transaction, combine that with the `sha256(txChangesBefore, operations, txChangesAfter)` and `sha256(txResult)`, which can both just be pulled instead of calculated, and then verify the hash of the combined hashes. 

If events were in OperationMeta, `hashes` [here](https://github.com/stellar/stellar-xdr/blob/1a76201f837e5b0f4fa5aff228ee865bfc1f6361/Stellar-ledger.x#L426) would contain `sha256(txChangesBefore, txChangesAfter)`, `sha256(operations)`, and `sha256(txResult)`. This would mean that the user would also need to hash all ledger entry changes to verify an event.